### DISCUSSION TITLE
Database design and UI Setup

### DISCUSSION THREAD

**Discussion by taholmes160** [Budibase/budibase#8666] [Help]
Hi Folks:

First off, this is a learning project for me, although we will also be using it at work, im new to budibase and to the internal DB

Here are some general notes about the database that I have come up with.
https://docs.google.com/document/d/1gSqmqD1guWXTGv5lf1i9HXwKeY0l7n3C_fCyjrgdpaY/edit?usp=sharing

I need help figuring out the best way to set up the database and tables, and how to make the UI the most user friendly and easiest to use both to avoid user frustration and to keep the data as clean as possible without miss spellings and other stuff that usually contaminates data

I am available most of the time on discord as well -- im in the budibase discord  https://discord.gg/VnQ6UNnj -- my user name is taholmes160

I could really use some help
Thanks
TIM

**Comment by melohagan** [comment]
Hey @taholmes160 

It may be a little more work as you'll need to maintain your own DB, however I think using a SQL database would be safest, such as Postgres.

The external databases support the auto-generated screens as well: https://docs.budibase.com/docs/screens#autogenerated-screens
whilst providing more flexibility with custom queries and handling relationships.  

With that said, I believe the internal Budibase DB could do everything you are trying to do, I just don't think it's the most suitable in this case. If requirements change, you may be more 'boxed-in'.
